app/src/main/java/com/example/sharingang/utils/PermissionRequest.kt

```diff
@@ -0,0 +1,54 @@
+package com.example.sharingang.utils
+
+import android.content.Context
+import android.content.pm.PackageManager
+import android.widget.Toast
+import androidx.activity.result.ActivityResultLauncher
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.content.ContextCompat
+import androidx.fragment.app.Fragment
+
+fun requestPermissionLauncher(
+    fragment: Fragment,
+    callback: () -> Unit
+): ActivityResultLauncher<String> {
+    return fragment.registerForActivityResult(ActivityResultContracts.RequestPermission()) { isGranted: Boolean ->
+        if (isGranted) {
+            callback()
+        }
+    }
+}
+
+fun doOrGetPermission(
+    context: Context?,
+    fragment: Fragment,
+    permission: String,
+    callback: () -> Unit,
```

Comment:
Maybe instead of just having a callback with no arguments, you could instead get the fine location inside of this function, and then have the callback take the location.

Well, maybe not inside of `doorGetPermission`, but you could make another wrapper specifically for the location, right.

Reply:
If you had a callback that took in the location, then that would simplify the duplication between both `updateLocation` methods, since you would just have the fragment specific code consuming that location.

Reply:
Ok I see, it makes sense!

Reply:
Actually, I wanted this function to be as general as possible, to be able to use it for all the methods that need certain permissions. For example, MapFragment needs to continuously get the location, whereas NewItem and EditItem only need it once.

Reply:
Yeah, my point was to make another wrapper around this function which would take a more specific callback.